### PR TITLE
Detect if base image uses buildpack-deps

### PR DESCRIPTION
**Pull request by retrohacker**

### CODE REVIEW

**Comment by retrohacker** [comment]
Take a look at `django`. It is built off of `python`, although `python` is built off of `buildpack-deps`. Should we be showing _variant.md_ or _variant-buildpack.md_ for this case?

**Comment by tianon** [comment]
I think for now, this is probably good enough.  I don't like that we're duplicating manifest file parsing logic outside bashbrew again, but I suppose we don't really have much choice given the constraints.

I've got a few stylistic comments for `variant.sh`, but overall I think this is good and gets us _something_. :+1:

File: .template-helpers/variant.sh [hunk 1]
```diff
@@ -9,9 +9,10 @@ if [ -z "$repo" ]; then
 fi
 
 dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+url='https://raw.githubusercontent.com/docker-library/official-images/master/library/'"$repo" 
 
 IFS=$'\n'
-tags=( $(curl -sSL 'https://raw.githubusercontent.com/docker-library/official-images/master/library/'"$repo" | grep -vE '^$|^#' | cut -d':' -f1 | sort -u) )
+tags=( $(curl -sSL $url | grep -vE '^$|^#' | cut -d':' -f1 | sort -u) )
```
Comment (tianon) [inline .template-helpers/variant.sh]: Can we please quote `$url` here? `curl -sSL "$url" ...`

File: .template-helpers/variant.sh [hunk 2]
```diff
@@ -23,9 +24,16 @@ for tag in "${tags[@]}"; do
 	fi
 done
 if [ "$text" ]; then
-	# give a little space
-	echo
-	echo
-	cat "$dir/variant.md"
-	echo "$text"
+  latest=($(curl -sSL $url | sed -e 's/git:\/\/github.com\///' -e 's/@/ /' - | grep "latest"))
```
Comment (tianon) [inline .template-helpers/variant.sh]: Same comment here (`curl -sSL "$url" ...`) -- also, that `\/\/` looks hairy.  How about something like `sed -e 's!git://github.com/!!' ...` instead?

Comment (tianon) [inline .template-helpers/variant.sh]: Also, can we please stick to tabs for indentation? :smile:

File: .template-helpers/variant.sh [hunk 3]
```diff
@@ -23,9 +24,16 @@ for tag in "${tags[@]}"; do
...
+  dockerfile='https://raw.githubusercontent.com/'"${latest[1]}"'/'"${latest[2]}"'/'"${latest[3]}"'/Dockerfile'
+  base_image=$(curl -sSL $dockerfile | sed 's/:/\t/' | nawk '/^FROM .*$/ { print $2 }')
```
Comment (tianon) [inline .template-helpers/variant.sh]: Same comment on quoting (`curl -sSL "$dockerfile" ...`), and I'd also note that we stick to `camelCasing` for variable names in Bash scripts.  As a final note, that last `awk` could be simplified as `awk '$1 == "FROM" { print $2 }'`.

**Comment by tianon** [comment]
That's all I see -- @yosifkit @Moghedrin, mind doing some review here too? :+1:

File: .template-helpers/variant.sh [hunk 4]
```diff
@@ -23,9 +24,16 @@ for tag in "${tags[@]}"; do
 	fi
 done
 if [ "$text" ]; then
-	# give a little space
-	echo
-	echo
-	cat "$dir/variant.md"
-	echo "$text"
+  latest=($(curl -sSL $url | sed -e 's/git:\/\/github.com\///' -e 's/@/ /' - | grep "latest"))
+  dockerfile='https://raw.githubusercontent.com/'"${latest[1]}"'/'"${latest[2]}"'/'"${latest[3]}"'/Dockerfile'
```
Comment (yosifkit) [inline .template-helpers/variant.sh]: I know there is [one place](https://github.com/infosiftr/docker-library-docs/blob/master/.template-helpers/generate-dockerfile-links-partial.sh#L52) where we have an assumption on it being a github repo. but that may not always be the case.  I just want to make sure we are aware of this possible problem later (especially if [pypy](https://bitbucket.org/pypy/pypy) wants to maintain their image).

Comment (tianon) [inline .template-helpers/variant.sh]: Oh, above that we even skip for non-github: https://github.com/infosiftr/docker-library-docs/blob/master/.template-helpers/generate-dockerfile-links-partial.sh#L30-L33

**Comment by yosifkit** [comment]
Other than my one comment and @tianon's comments for changes, that should work for now.

File: .template-helpers/variant.sh (hunk 2 above)
Comment (yosifkit) [inline .template-helpers/variant.sh]: We should also keep in mind that the format here will probably change sometime after 1.6 is released: https://github.com/docker-library/official-images/issues/549

File: .template-helpers/variant.sh (hunk 4 above)
Comment (retrohacker) [inline .template-helpers/variant.sh]: I changed the grep to `grep "latest.*github.com"` and then do a `if [ -z "$latest" ]; then exit 0; fi`

**Comment by yosifkit** [comment]
LGTM

**Comment by tianon** [comment]
Sorry for the delay; this is definitely an improvement.  It's not as clean a solution as I'd like, but `something > nothing`, and I don't have better ideas, so :+1:.

LGTM

**Pull request merged**